src/shared/serialize.c

```diff
@@ -197,9 +197,9 @@ int serialize_pidref(FILE *f, FDSet *fds, const char *key, PidRef *pidref) {
                 if (copy < 0)
                         return log_error_errno(copy, "Failed to add file descriptor to serialization set: %m");
 
-                r = serialize_item_format(f, key, "@%i", copy);
+                r = serialize_item_format(f, key, "@%i:" PID_FMT, copy, pidref->pid);
```

Comment:
can you add a comment here that explains why we serialize both?

i.e. `/* we serialize pidfd *and* pid, since a pidfd to a reaped process won't store the pid the process used to have */